head.S

```diff
@@ -119,7 +120,13 @@ GLOBAL(_entry)
 1:
 #endif	/* 64bit setup. */
 
+	cmpl	$0, boot_protocol(%ebp)
+	jne	1f
 	call	lz_main
+	jmp	2f
+1:
+	call	lz_multiboot2
+2:
```

Comment:
As we develop features/functionality, it will be far easier to deal with things in a single main function, especially when we get DEV/IOMMU protections.  I'd recommend having a `switch (boot_protocol)` or `if/else` chain in the middle of lz_main, rather than having a totally separate entrypoint

Reply:
Done